### PR TITLE
Run tests as a github check on every PR

### PR DESCRIPTION
**Pull request by WcaleNieWolny**
Fix https://github.com/Cap-go/capgo/issues/424

@ologbonowiwi can you please review?

/claim https://github.com/Cap-go/capgo/issues/424

### CODE REVIEW

**Comment by ayewo** [comment]
@ayewo here 👋

I'm going over the code but could you briefly describe how the new implementation works? Or even better, it would be nice to update the README describing how the bot works/does when triggered.

**Comment by WcaleNieWolny** [comment]
There is a thing called check suite in GitHub API that triggers on EVERY commit. I just use that event + events for requesting a rerun to start a workflow and add a status to a commit (say that the workflow is in progress/running/has failed etc). This workflow then updates this commit status based on the output. I removed the /test the command in favor or running just this workflow. This workflow will automatically start for repos where this GitHub all is installed.

I will change contributing.md in capgo to include this changed bot workflow

**Comment by ayewo** [comment]
But it's only configured to work with the main Capgo repo and the CLI repo, correct?

(I noticed that the original code before your changes in this PR doesn't call [`pn run test_rls`](https://github.com/Cap-go/CLI/blob/894d99a9be112df82e21b9f4d9e2055363cfc134/package.json#L39C6-L39C14) for the CLI repo.)

File: .github/workflows/test_cli.yml [hunk 1]
```diff
@@ -40,27 +49,30 @@ jobs:
...
-                body: ':hammer_and_wrench: Test\n\n:arrows_counterclockwise: status: Running :runner:\n:robot: CI/CD run is available [here](CICD_RUN_URL)'.replace('CICD_RUN_URL', jobUrl),
+            github.rest.checks.update({
+              owner: '${{ inputs.repo_owner }}',
+              repo: '${{ inputs.repo_name }}',
+              name: 'E2E tests',
+              status: 'in_progress',
+              head_sha: '${{ inputs.commit_sha }}',
+              details_url: jobUrl,
+              output: {
+                summary: `The detalided output is available [here](${jobUrl})`,
```
Comment (ayewo) [inline .github/workflows/test_cli.yml:74]: @WcaleNieWolny can you please enable the checkbox: "Allow edits and access to secrets by maintainers" ?
image


I get a 403 when I try to commit some typo fixes to your branch:
```sh
remote: Permission to WcaleNieWolny/capgo-github-bot.git denied to ayewo.
fatal: unable to access 'https://github.com/WcaleNieWolny/capgo-github-bot/': The requested URL returned error: 403
```

Comment (ayewo) [reply .github/workflows/test_cli.yml:74]: I figured it was faster to fix them than to manually point them out, one by one.

**Comment by WcaleNieWolny** [comment]
I do not care about testing rls. If that is not added in the testing framework then it is not the job of this workflow to call it

If you want to test rls headers then add it to capgo main repo (CLI tests) but DO NOT add it to this workflow

**Comment by WcaleNieWolny** [comment]
There are a lot of different testing frameworks used in capgo (playwright, deno tests, custom fetch). It is the job of the custom test runner to integrate them all. The job of this workflow is to call only this custom testing framework

File: .github/workflows/test_cli.yml [hunk 2]
```diff
@@ -40,27 +49,30 @@ jobs:
...
-            console.log(job)
+
             const jobId = job.data.jobs[0].id
             if (!jobId) {
               throw new Error('No job id')
             }
 
             const jobUrl = 'https://github.com/Cap-go/github-bot/actions/runs/RUN_ID/job/RUN_NUMBER'.replace('RUN_ID', context.runId).replace('RUN_NUMBER', jobId)
-            core.exportVariable('JOB_URL', jobUrl)
+            core.exportVariable('JOB_URL', jobUrl)
```
Comment (ologbonowiwi) [inline .github/workflows/test_cli.yml:64]:
```suggestion
            core.exportVariable('JOB_URL', jobUrl)
```

File: .github/workflows/test_cli.yml (hunk 1 above)
Comment (ologbonowiwi) [inline .github/workflows/test_cli.yml:74]:
```suggestion
                summary: `The detailed output is available [here](${jobUrl})`,
```

File: .github/workflows/test_cli.yml [hunk 3]
```diff
@@ -154,11 +166,17 @@ jobs:
...
+            github.rest.checks.create({
+              owner: '${{ inputs.repo_owner }}',
+              repo: '${{ inputs.repo_name }}',
+              name: 'E2E tests',
+              status: 'completed',
+              conclusion: 'failure',
+              head_sha: '${{ inputs.commit_sha }}',
+              details_url: JOB_URL,
+              output: {
+                summary: `The detalided output is available [here](${JOB_URL})`,
```
Comment (ologbonowiwi) [inline .github/workflows/test_cli.yml:178]:
```suggestion
                summary: `The detailed output is available [here](${JOB_URL})`,
```

File: .github/workflows/test_cli.yml [hunk 4]
```diff
@@ -169,11 +187,17 @@ jobs:
...
+            github.rest.checks.update({
+              owner: '${{ inputs.repo_owner }}',
+              repo: '${{ inputs.repo_name }}',
+              name: 'E2E tests',
+              status: 'completed',
+              conclusion: 'success',
+              head_sha: '${{ inputs.commit_sha }}',
+              details_url: JOB_URL,
+              output: {
+                summary: `The detalided output is available [here](${JOB_URL})`,
```
Comment (ologbonowiwi) [inline .github/workflows/test_cli.yml:199]:
```suggestion
                summary: `The detailed output is available [here](${JOB_URL})`,
```

**Review by ologbonowiwi — APPROVED** [review]
the PR implementation LGTM, but honestly I see this approach as quite complex to have tests on the repo.

It'd be easier to use "standard github actions" rather than do all this by ourselves, you can see in [this example](https://github.com/tailcallhq/tailcall/blob/main/.github/workflows/ci.yml#L26-L61) what I mean by standard github action.

**Comment by WcaleNieWolny** [comment]
Merging this patch was way more difficult then expected. @ayewo do you know if I have done anything wrong?

image

**Comment by ayewo** [comment]
> Merging this patch was way more difficult then expected. @ayewo do you know if I have done anything wrong?
> 
> image

Sorry for the trouble. The command looks correct so no idea what may have gone wrong. 
Nonetheless I see you were able to apply the patch 👍.

**Comment by ayewo** [comment]
Other than the README / CONTRIBUTING which you've promised to update, LGTM!

Well done @WcaleNieWolny 🚀!

**Comment by riderx** [comment]
@ologbonowiwi thanks for the comment, I do agree with you a GitHub action was looking nicer, the problem is when we do have changed in the CLI and Capgo, that what address the work of @WcaleNieWolny we can set the test to run on certain branch.
@ayewo for the `test_rls` in the CLI folder they were useless so that ok, i believe @WcaleNieWolny test are doing a better job.

**Pull request merged**